pylint/extensions/code_style.py

```diff
@@ -309,12 +309,13 @@ def _check_ignore_assignment_expr_suggestion(
         if isinstance(node.test, nodes.Compare):
             next_if_node: nodes.If | None = None
             next_sibling = node.next_sibling()
-            if len(node.orelse) == 1 and isinstance(node.orelse[0], nodes.If):
-                # elif block
-                next_if_node = node.orelse[0]
-            elif isinstance(next_sibling, nodes.If):
-                # separate if block
-                next_if_node = next_sibling
+            match (node.orelse, next_sibling):
+                case [[nodes.If() as next_if_node], _]:
+                    # elif block
+                    pass
+                case [_, nodes.If() as next_if_node]:
+                    # separate if block
+                    pass
```

Comment:
Not sure about this one. It works 🤷🏻‍♂️ There are a few things to note though.

**Only variable binding - no body**
So far I've seen quite a few cases where it might make sense to "abuse" match to just do the variable binding. I considered using `...` as body to make it more obvious that's what's happing but that doesn't look good / feels wrong. Maybe just `pass` is still the best option.

**Performance**
For the match to work here, I matched each case against a constructed sequence. The plain if-elif is probably a bit faster though I haven't profiled it yet. Ideally something like a match expression would be added to Python at some point. Something like
```py
if node.orelse match [nodes.If() as next_if_node]:
    pass
elif isinstance(next_sibling, nodes.If):
    ...
```